plugins/google/sql/sqlContainedDatabaseAuth.js

```diff
@@ -0,0 +1,68 @@
+    more_info: 'SQL instance for SQLServer databases provides Contained Database Authentication flag. It is used to shift the user authentication process from database engine level to database level. It is enabled by default and should always be disabled to keep the authentication enabled for security.',
+    link: 'https://cloud.google.com/sql/docs/sqlserver/flags',
+    recommended_action: 'Ensure that Contained Database Authentication flag is disabled for all SQLServer instances.',
+    apis: ['instances:sql:list'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions();
+
+        async.each(regions.instances.sql, function(region, rcb) {
+            let sqlInstances = helpers.addSource(
+                cache, source, ['instances', 'sql', 'list', region]);
+
+            if (!sqlInstances) return rcb();
+
+            if (sqlInstances.err || !sqlInstances.data) {
+                helpers.addResult(results, 3, 'Unable to query SQL instances: ' + helpers.addError(sqlInstances), region);
+                return rcb();
+            }
+
+            if (!sqlInstances.data.length) {
+                helpers.addResult(results, 0, 'No SQL instances found', region);
+                return rcb();
+            }
+
+            sqlInstances.data.forEach(sqlInstance => {
+                if (sqlInstance.databaseVersion && !sqlInstance.databaseVersion.toLowerCase().includes('sqlserver')) {
+                    helpers.addResult(results, 0, 
+                        'SQL instance database type is not of SQLServer type', region, sqlInstance.name);
+                    return;
+                }
+
+                if (sqlInstance.instanceType != "READ_REPLICA_INSTANCE" &&
+                    sqlInstance.settings &&
+                    sqlInstance.settings.databaseFlags &&
+                    sqlInstance.settings.databaseFlags.length) {
+                        let found = sqlInstance.settings.databaseFlags.find(flag => flag.name && flag.name == 'contained database authentication' &&
+                                                                        flag.value && flag.value == 'off');
+
+                        if (found) {
+                            helpers.addResult(results, 0, 
+                                'SQL instance does not have contained database authentication flag enabled', region, sqlInstance.name);
+                        } else {
+                            helpers.addResult(results, 2,
+                                'SQL instance has contained database authentication flag enabled', region, sqlInstance.name);
+                        }
+                } else if (sqlInstance.instanceType == "READ_REPLICA_INSTANCE") {
+                } else {
+                    helpers.addResult(results, 2, 
+                        'SQL instance has contained database authentication flag enabled', region, sqlInstance.name);
```

Comment:
this should pass if there are no db flags

Reply:
Updated.